themes/barbuk/README.md

```diff
@@ -9,6 +9,14 @@ A minimal theme with a clean git prompt
 * Current git info
 * Last command exit code (only shown when the exit code is greater than 0)
 
+## Font
+
+A font with SCM glyphs is required to display the default tool/host logos.
+You can use a font from https://www.nerdfonts.com/ or patch your own font with the tool
+provided by https://github.com/ryanoasis/nerd-fonts.
+
+You can also override the default variables if you want to use different glyphs or standard ASCII characters.
```

Comment:
Aha !  I just realized I was wrong about being able to override the default variables in this theme - My apologies.

If you want your theme to be customizable by the end-user, you could use `"conditional assignments"` for your variables.

Taking the Powerline theme as an example, the usage here might be something like:
```
SCM_GIT_CHAR_GITLAB=${BARBUK_SCM_GITLAB_CHAR:="'  '}
```

The user could set `BARBUK_SCM_GITLAB_CHAR` within their .bash_profile in order to control what character(s) is used for the logo.

Those kind of changes are beyond this PR's intention so I would not expect you to do that here.   So removing this line is probably the correct thing to do for now.

Again, my apologies.

-DF

Reply:
> Those kind of changes are beyond this PR's intention so I would not expect you to do that here. So removing this line is probably the correct thing to do for now.

It's a nice suggestion, and I think it's completely in the PR scope (maybe not the initial attention), because people may not want to install a nerdfont and change the glyphs.

I will make some change so that people can better customize the theme to their liking.

Reply:
I've added the option to override default variables.

I've kept the same variable name as to not introduce a specific variable for the theme, so if another theme want to use the variable, the user customization can stay the same:

```bash
SCM_GIT_CHAR_GITLAB=${SCM_GITLAB_CHAR:='  '}
```

Is this an accepted practice for the project ?

Reply:
> I think it's completely in the PR scope

You convinced me :)

> Is this an accepted practice for the project ?

I can think of a few reasons why you want to use theme-specific names for the user-defined variables, but the primary reason is:

* You cannot use conditional-assignment on variables already defined bash-it init scripts

Have a look at [base.theme.bash](https://github.com/Bash-it/bash-it/blob/159a1ce3e05255c27a0dfe8a97046eca4d0b5715/themes/base.theme.bash) - Because these variables are already assigned, a theme cannot apply their default values to them using conditional-assignment.  Having a theme-specific name for the user variable creates the ability for a theme to both have their theme-specific  default values and allow the user to override them if desired.

Another issue that comes to mind:

*  Scope Consistency

A variable that *looks* like a global variable, but is really only specific to your theme can be a bit confusing if exposed to the user.   I think this applies to **all** of your `SCM_GIT_*` variables - They *look* like they could apply to the global scm functions but really they are only local to your theme.

**NOTE:** Of course its okay for your theme to use these variables internally - I'm just advocating that it could be confusing for the user to use these variables in their bash_profile.

So my suggestion is to use a theme prefix  (and simplified suffix) for the user-defined variables.  Maybe something like:

```
# Theme custom glyphs
SCM_GIT_CHAR_GITLAB=${BARBUK_GITLAB_CHAR:='  '}
SCM_GIT_CHAR_BITBUCKET=${BARBUK_BITBUCKET_CHAR:='  '}
SCM_GIT_CHAR_GITHUB=${BARBUK_GITHUB_CHAR:='  '}
SCM_GIT_CHAR_DEFAULT=${BARBUK_GIT_DEFAULT_CHAR:='  '}
SCM_GIT_CHAR_ICON_BRANCH=${BARBUK_BRANCH_ICON:=''}
EXIT_CODE_ICON=${BARBUK_EXIT_CODE_ICON:=' '}
```

Hope that helps, let me know what you think !

-DF

PS: I'm just a drive-by reviewer, not a repo maintainer, so these bites are just opinions, not requirements.

Reply:
Thanks for taking the time to provide a detailed feedback.

Your argument about scope consistency was great, I was going to follow it, but the PR is now merged.

Reply:
Sorry, missed that part of the conversation. If it's an important change, please feel free to open another PR...

Reply:
That is all right, it's not an important change, more of a conversation about var naming and scope.

I will think about it making another PR if needed.